### DISCUSSION TITLE
Bug Report

### DISCUSSION THREAD

**Discussion by Draconis13** [FakeFishGames/Barotrauma#14683] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Multiple issues with creature spawns while working on Barotraumatic and the Creature Pack  since the last update when modding in creature event sets. Primary issue is creatures are spawning at level start and the game is not taking into account the mindistance travelled and/or mintime along with the allowatstart.  This wasn't a problem before and I have been taking a lot of time to try to fix this. In testing it seems to work fine for the most part but I get reports from numerous server of odd behavior where too many spawns overlap, spawns occur at level start before sub moves, massive amounts of spawns after hour and a half (yes that is a long time but still). 

I have tried to narrow it down to the usual cave, wreck, wreck encounter and abyss with the Barotraumatic spawn events of distance 20,40,60,80 and the passing fish main+side and passing fish abyss spawns. However certain servers or campaigns seem to bug out and I joined one last night where it looks like all the distance events occur at the same time causing a huge monster mash fight at the start and the rest of the level is empty except for the occasional passing fish event. I do not know if this is a change in how these arguments work to influence the event set or a straight up bug. 

 
    <EventSet chooserandom="true" mindistancetraveled="0.15" minmissiontime="400" allowatstart="false" triggereventcooldown="true" ignorecooldown="false" delaywhencrewaway="false" resettime="960">
   

    
    <EventSet chooserandom="true" mindistancetraveled="0.20" allowatstart="false" ignorecooldown="false" triggereventcooldown="true" >
      
        
    
<!--	<EventSet chooserandom="true" mindistancetraveled="0.33" minmissiontime="1080" allowatstart="false" triggereventcooldown="true" ignorecooldown="false" delaywhencrewaway="false" resettime="1500">

### Reproduction steps

Start and play in a campaign with Barotraumatic and Barotraumatic creature pack. At some point the monster spawns seem to break and cause unexpected and unintended behavior such as all spawns occurring at start of level once the sub moves slightly problem seems to persist until end of campaign if it occurs. 

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer hosted using a dedicated server

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Have not encountered it in either singleplayer or multiplayer testing runs but have joined servers with similar modlists and the issue persist and is often a constant and predictable as clockwork occurrence, have not seen any errors in console regarding event timers or event sets so its not producing an error of any kind that I am aware of.
```

**Comment by A2ur3** [comment]
i did not encounter these issues in my custom eventset
dont use resettime #13871 #13872 #13873 
if you want to make monsters spawn every x seconds you need to copy and paste y times 💀

**Comment by Draconis13** [comment]
Hmm I think you might be right on that, It was here when I took over and it seemed functional but the more I look into it their seems to be a problem with the reset to try to make the event spawn again, I think triple flight of passing fish events will solve this

**Comment by Draconis13** [comment]
The passing fish event was supposed to work similar to the abyss spawns where it would over time provide a number of small threats over the course of the mission so that there was never any 100% safe times for the players to game as is the issue with abyss. I.E. Kill abyss monster reset time is X or disabled due to hunting grounds allowing free exploration of the map, previously some events had 10000 seconds on them and were formatted incorrectly and never actually went off even for groups who spent 2 hours in the mission but as of last patch these old events would trigger on and spawn endlessly after 10000 seconds causing massive problems and so many creatures servers would lag out. I fixed those portions but getting a reliable long duration timer similar to abyss has proven to be a huge issue. 

The problem though isn't from the passing fish events and their reset times but something to do with the 20,40,60,80 set all spawning at the same time at the start of level. Possibly crossed wires with the two fish and abyss resets? 

I am going to try a triple pack of  the following for fish with mindistance and time and see if that finally fixes it. 

passing fish 0.15 400, 0.50 1200, 0.70 1800

abyss fish 0.25 500, 0.55 1500, 0.75 2300

**Comment by Draconis13** [comment]
![image](https://github.com/user-attachments/assets/030725f8-df0e-4868-8dc9-dfac5e049bf4)

So here is a screenshot of testing the new passing fish/abyss spawns in mission mode with a fixed mission mode for BT (next patch) I have three spawns clustered around the start of the level  while two(!) wrecks right below me have their proper wreck/encounter as per normal

**Comment by Draconis13** [comment]
![image](https://github.com/user-attachments/assets/8ab8f3bf-fefa-4181-9cf3-4a533df3a169)
After several attempts with the new arguements and triple fish + triple abyss it constantly spawns 3 right at the start and I cannot figure out what is going wrong here. They are all mainpath/sidepath events from the mod.  

It seems like it is ignoring mindistance on several of the distance events and consistently spawning them right at the start of the game.

**Comment by A2ur3** [comment]
mindistancetraveled         = 0-1. How large portion of the level must the submarine traverse before the event can be activated.
minmissiontime              = How many seconds must have passed before the event can be activated. Note that the event activates
if either mindistancetraveled OR minmissiontime is reached. If you want to make the event purely time-based or only distance-based, you can set the distance or time limit to a very high value.

**Reply by Draconis13** [reply]
Testing it seems the 20,40,60,80 events had been lacking the eventcount and due to that the first event 20 was firing off the other 3? seems confusing but I tested it a few times now and the events are going correctly. 

I also identified all of the event sets so maybe that will help them perhaps not get triggered at the same time?

Also yeah its supposed to have them go off if you wait around too long in an area on the passing fish events. the 20-80 are all distance based. It seems to be working a lot better now I might add back in the timed event with a long timer but that event has been exceptionally cursed and caused spawning on level start without fail but with a lot of its arguements removed perhaps it could be brought back in as the long time in level spawn? will have to see.
